001-core/password.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+add_action( 'user_profile_update_errors', 'validate_current_password', 1, 3 );
+
+function validate_current_password( $errors, $update, $user ) {
+	check_admin_referer( 'update-user_' . $user->ID );
+
+	if ( ! isset( $_POST['pass1'] ) || empty( $_POST['pass1'] ) || ! $update ) {
+		return;
+	}
+
+	if ( ! isset( $_POST['current_pass'] ) || empty( $_POST['current_pass'] ) ) {
+		$errors->add( 'wrong_current_password', __( '<strong>ERROR</strong>: Please enter your current password.' ), array( 'form-field' => 'current_pass' ) );
+		return;
+	}
+
+	$error = wp_authenticate( $user->user_login, $_POST['current_pass'] );
+	if ( is_wp_error( $error ) ) {
+		$errors->add( 'wrong_current_password', __( '<strong>ERROR</strong>: The entered current password is not correct.' ), array( 'form-field' => 'current_pass' ) );
+	}
+}
+
+add_action( 'show_user_profile', 'add_current_password_field' );
+
+function add_current_password_field() { ?>
+	<table id="nojs-current-pass" class="form-table" role="presentation">
+		<tr>
+			<th><label for="current_pass"><?php _e( 'Current Password' ); ?></label></th>
```

Comment:
Does this repo treat future translated strings as trusted? If not, then `_e` should be `esc_html_e()` and other translated strings need escaping too.

Reply:
Not sure about this one. I only found `esc_html_e` inside the Akismet folder, so I'd like to double check if that is a repo-wide thing.

Reply:
Let's use the escaped versions when possible. It would be difficult, but not impossible for an attack to be carried out through a translation. Thanks for the reminder @GaryJones!